guides/javascript.md

```diff
@@ -0,0 +1,176 @@
+# JavaScript
+
+**NOTE:** All of these guidelines apply to TypeScript as well.
```

Comment:
So there are some guidelines which apply to JavaScript and TypeScript, and some which apply only to TypeScript. I wasn't sure how best to represent this. I figured that anything applicable to JavaScript was also applicable to TypeScript (but not vice versa) but I'm not sure if that's immediately obvious or not.